rio_tiler/mosaic/reader.py

```diff
@@ -59,12 +60,18 @@ def reader(asset: str, *args, **kwargs) -> Tuple[numpy.ndarray, numpy.ndarray]:
         Return (tile, mask) data and list of assets used.
 
     """
+    if isclass(pixel_selection):
+        pixel_selection = cast(Type[MosaicMethodBase], pixel_selection)
```

Comment:
Not sure to understand why we need this!

Reply:
Mypy errors without this because the input can be either a class or an instance and it fails on an instance. Mypy isn't smart enough to understand the `isclass and`

Reply:
So it's telling mypy that within the if block, it's a class not an instance

Reply:
what do you think about adding `# type: ignore` instead of this new if block ?

Reply:
It's roughly equivalent but the `cast` is more explicit; it's still checking instead of just ignoring

Reply:
Ok, at this point I think it might be better to pass a class instead of an object 🤷‍♂️ (this will be a breaking change but it might be cleaner)